src/date/expand-pattern.js

```diff
@@ -33,10 +55,30 @@ return function( pattern, cldr ) {
 
 	switch ( true ) {
 		case "skeleton" in pattern:
+			skeleton = pattern.skeleton;
 			result = cldr.main([
 				"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-				pattern.skeleton
+				skeleton
 			]);
+			if ( result === undefined ) {
+				for ( i = 0 ; i < skeleton.length ; i++ ) {
+					if ( /[hHms]/.test( skeleton[i] ) ) {
+						timeSkeleton += skeleton[i];
+					} else if ( /[GyQMEd]/.test( skeleton[i] ) ) {
+						dateSkeleton += skeleton[i];
+					}
+				}
```

Comment:
Note this approach may lead to undesired results. For example, take this invalid skeleton `MhMhMh` as an input. We would wrongly have timeSkeleton `hhh` and dateSkeleton `MMM`.

One way to fix that is using the below approach:

```js
timeSkeleton = skeleton.split( /[^hHms]/ ).slice( -1 )[ 0 ];
```

PS: Note I'm using `hHms`, but it should obviously be extended with all the correct time fields.

PS2: For dateSkeleton, `slice` is not needed.